common/src/main/java/com/ably/tracking/common/DefaultAblySDK.kt

```diff
@@ -0,0 +1,170 @@
+package com.ably.tracking.common
+
+import io.ably.lib.realtime.AblyRealtime
+import io.ably.lib.realtime.Channel.MessageListener
+import io.ably.lib.realtime.ChannelState
+import io.ably.lib.realtime.ChannelStateListener
+import io.ably.lib.realtime.CompletionListener
+import io.ably.lib.realtime.ConnectionState
+import io.ably.lib.realtime.ConnectionStateListener
+import io.ably.lib.realtime.Presence.PresenceListener
+import io.ably.lib.rest.Auth.RenewAuthResult
+import io.ably.lib.types.ChannelOptions
+import io.ably.lib.types.ClientOptions
+import io.ably.lib.types.ErrorInfo
+import io.ably.lib.types.Message
+import io.ably.lib.types.PresenceMessage
+
+/**
+ * An implementation of [AblySDKRealtimeFactory] which uses the `ably-java` client library.
+ */
+class DefaultAblySDKRealtimeFactory : AblySDKRealtimeFactory {
+    override fun create(clientOptions: ClientOptions): AblySDKRealtime {
+        return DefaultAblySDKRealtime(clientOptions)
+    }
+}
+
+class DefaultAblySDKRealtime
+constructor(clientOptions: ClientOptions) : AblySDKRealtime {
+    private val realtime = AblyRealtime(clientOptions)
+
+    override val auth: AblySDKRealtime.Auth
+        get() = Auth(realtime.auth)
```

Comment:
When you use the `get()`it creates a getter method that is called each time someone accesses the `val`. Therefore, whenever you will call `DefaultAblySDKRealtime.auth` it will create a new `Auth` instance. It's better to simply set the `val`'s value instead of using the `get()` method in this case. I've made [a simple code example](https://pl.kotl.in/umm77Af__) that should show the difference 😉 
```suggestion
    override val auth: AblySDKRealtime.Auth = Auth(realtime.auth)
```
The above change should be applied to all properties that should be created only once.

Reply:
Thanks for that, I've made that change 👍